packages/db-objects-file/lib/objects/objectsInMemServerRedis.js

```diff
@@ -280,57 +279,43 @@ class ObjectsInMemoryServer extends ObjectsInMemoryFileDB {
                         this.log.silly(`${namespaceLog} Script transformed into getObjectView: design=${scriptDesign}, search=${scriptSearch}`);
                     }
-                    this._getObjectView(scriptDesign, scriptSearch, {
-                        startkey: data[3],
-                        endkey: data[4],
-                        include_docs: true
-                    }, (err, objs) => {
-                        if (err) {
-                            handler && handler.sendError(responseId, new Error('_getObjectView Error for ' + scriptDesign + '/' + scriptSearch + ': ' + err));
-                            return;
-                        }
-                        const res = [];
-
-                        objs.rows.forEach(obj =>
-                            res.push(JSON.stringify(this.dataset[obj.value._id || obj.id])));
-
-                        handler && handler.sendArray(responseId, res);
-                    });
-                    return;
+                    let objs;
+                    try {
+                        objs = this._getObjectView(scriptDesign, scriptSearch, {
+                            startkey: data[3],
+                            endkey: data[4],
+                            include_docs: true
+                        });
+                    } catch (err) {
+                        return void handler.sendError(responseId, new Error('_getObjectView Error for ' + scriptDesign + '/' + scriptSearch + ': ' + err));
+                    }
+                    const res = objs.rows.map(obj => JSON.stringify(this.dataset[obj.value._id || obj.id]));
+                    handler.sendArray(responseId, res);
                 }
             } else if (this.knownScripts[data[0]].func && data.length > 4) {
                 const scriptFunc = {map: this.knownScripts[data[0]].func.replace('%1', data[5])};
                 if (this.settings.connection.enhancedLogging) {
                     this.log.silly(`${namespaceLog} Script transformed into _applyView: func=${scriptFunc.map}`);
                 }
-                this._applyView(scriptFunc, {
+                const objs = this._applyView(scriptFunc, {
                     startkey: data[3],
                     endkey: data[4],
                     include_docs: true
-                }, (err, objs) => {
-                    if (err) {
-                        handler && handler.sendError(responseId, new Error('_getObjectView Error ' + err));
-                        return;
-                    }
-                    const res = [];
-
-                    objs.rows.forEach(obj =>
-                        res.push(JSON.stringify(this.dataset[obj.value._id || obj.id])));
-
-                    handler && handler.sendArray(responseId, res);
                 });
-                return;
+                const res = objs.rows.map(obj => JSON.stringify(this.dataset[obj.value._id || obj.id]));
+
+                return void handler.sendArray(responseId, res);
```

Comment:
I don't remember why we started doing this (there was probably a good reason for that). Just calling the method and then `return;` afterwards is just fine.

Reply:
And more readyble IMO 🤷🏻‍♂️

Reply:
Applies to a couple of locations

Reply:
Yes we just started some time ago ... it safes a line :-)